storage/backup/backup-cli/src/backup_types/transaction/tests.rs

```diff
@@ -120,19 +120,22 @@ fn end_to_end() {
     let ouptputlist = tgt_db
         .get_transaction_outputs(0, target_version, target_version)
         .unwrap();
-    assert_eq!(
-        ouptputlist
-            .transactions_and_outputs
-            .iter()
-            .map(|(_, output)| output.write_set().clone())
-            .collect::<Vec<_>>(),
-        blocks
-            .iter()
-            .flat_map(|(txns, _li)| txns)
-            .take(target_version as usize)
-            .map(|txn_to_commit| txn_to_commit.write_set().clone())
-            .collect::<Vec<_>>()
-    );
+
+    for (restore_ws, org_ws) in ouptputlist
+        .transactions_and_outputs
+        .iter()
+        .map(|(_, output)| output.write_set().clone())
+        .zip(
```

Comment:
why the change?
This is worse because it doesn't assert both sequences have the same length (for example, if either of them is empty, this will pass).